content/2024-berlin/talks/the-weirdest-promql-youll-ever-see-promql-for-reporting-analytics-and-business-intelligence.md

```diff
@@ -4,29 +4,28 @@ title: "The weirdest PromQL you’ll ever see: PromQL for Reporting, Analytics,
 
 ## The weirdest PromQL you’ll ever see: PromQL for Reporting, Analytics, and Business Intelligence
 
-Prometheus is fantastic at managing high frequency data. But what if you have an analytics use case? Sometimes we just want to lower the frequency; to see daily, weekly or monthly data and find long-term trends or growth rates.
+Prometheus is fantastic at managing high-frequency data. But what if you have an analytics use case? Sometimes we just want to lower the frequency; to see daily, weekly or monthly data and find long-term trends or growth rates.
 
-We might want to consider any of these use cases per day, week or month, such as:
+We might want to consider any of these use cases per day, week, or month, such as:
 - Activity: requests or transactions
-- Usage or consumption, for billing or auditing
+- Usage or consumption for billing or auditing
 - Count of events: alerts or incidents, for example.
 
 Why use Prometheus rather than more traditional analytics tools for this? By querying Prometheus directly we can avoid creating/maintaining a data-pipelines, avoid data exports and staleness problems, and lean on the power of the Prometheus ecosystem: For example, we can create overviews for management, plan ahead, and alert on our reports.
 
 Yet querying Prometheus by calendar-month in particular can be surprisingly tricky. PromQL doesn’t support using 1M in subqueries because month-lengths vary.
 
 In this talk I’ll:
-- Take you on a journey, to understand all the ways that the time dimension can be controlled when querying PromQL with:
-  - The @ operator 
-  - Subqueries
-  - The offset keyword
-- Navigate more PromQL gotchas along the way:
-  - Why do we need rate and label_replace
-  - How to control the MaxDataPoints when querying from Grafana
-  - Use of Joins and the absent function
-- Conclude by showing two different ways we can render monthly data FTW:
-  - Render a saw-tooth cumulative counter which resets to zero every month-start
-  - Use `and day_of_month()` in your query to get true monthly data from a counter
+
+* Take you on a journey, discovering ways that the time dimension can be aggregated when querying Prometheus by using:
+  * The offset keyword
+  * Joins and the absent function
+  * Union and intersection operators or and

```

Comment:
I'm not sure what flavour of markdown we render. I feel like Github flavour requires 3 spaces when indenting bullets. But maybe they reversed that decision - I remember that from years ago.

If 3 spaces are needed, then this would be
```suggestion
   * The offset keyword
   * Joins and the absent function
   * Union and intersection operators or and

```